src/algebra/ordered_monoid.lean

```diff
@@ -75,6 +75,70 @@ ordered_comm_monoid.mul_le_mul_left a b h c
 lemma mul_le_mul_right' (h : a ≤ b) (c) : a * c ≤ b * c :=
 by { convert mul_le_mul_left' h c using 1; rw mul_comm }
 
+@[to_additive]
+lemma mul_lt_of_mul_lt_left (h : a * b < c) (hle : d ≤ b) : a * d < c :=
+lt_of_le_of_lt (mul_le_mul_left' hle a) h
+
+@[to_additive]
+lemma lt_of_mul_lt_left (h : a * b < c) (hle : 1 ≤ b) : a < c :=
+by simpa only [mul_one] using mul_lt_of_mul_lt_left h hle
+
+@[to_additive]
+lemma mul_lt_of_mul_lt_right (h : a * b < c) (hle : d ≤ a) : d * b < c :=
+lt_of_le_of_lt (mul_le_mul_right' hle b) h
+
+@[to_additive]
+lemma lt_of_mul_lt_right (h : a * b < c) (hle : 1 ≤ a) : b < c :=
+by simpa only [one_mul] using mul_lt_of_mul_lt_right h hle
+
+@[to_additive]
+lemma mul_le_of_mul_le_left (h : a * b ≤ c) (hle : d ≤ b) : a * d ≤ c :=
+(mul_le_mul_left' hle a).trans h
+
+@[to_additive]
+lemma le_of_mul_le_left (h : a * b ≤ c) (hle : 1 ≤ b) : a ≤ c :=
+by simpa only [mul_one] using mul_le_of_mul_le_left h hle
+
+@[to_additive]
+lemma mul_le_of_mul_le_right (h : a * b ≤ c) (hle : d ≤ a) : d * b ≤ c :=
+(mul_le_mul_right' hle b).trans h
+
+@[to_additive]
+lemma le_of_mul_le_right (h : a * b ≤ c) (hle : 1 ≤ a) : b ≤ c :=
+by simpa only [one_mul] using mul_le_of_mul_le_right h hle
+
+@[to_additive]
+lemma lt_mul_of_lt_mul_left (h : a < b * c) (hle : c ≤ d) : a < b * d :=
+lt_of_lt_of_le h (mul_le_mul_left' hle b)
+
+@[to_additive]
+lemma lt_of_lt_mul_left (h : a < b * c) (hle : c ≤ 1) : a < b :=
+by simpa only [mul_one] using lt_mul_of_lt_mul_left h hle
+
+@[to_additive]
+lemma lt_mul_of_lt_mul_right (h : a < b * c) (hle : b ≤ d) : a < d * c :=
+lt_of_lt_of_le h (mul_le_mul_right' hle c)
+
+@[to_additive]
+lemma lt_of_lt_mul_right (h : a < b * c) (hle : b ≤ 1) : a < c :=
+by simpa only [one_mul] using lt_mul_of_lt_mul_right h hle
+
+@[to_additive]
+lemma le_mul_of_le_mul_left (h : a ≤ b * c) (hle : c ≤ d) : a ≤ b * d :=
+h.trans (mul_le_mul_left' hle b)
+
+@[to_additive]
+lemma le_of_le_mul_left (h : a ≤ b * c) (hle : c ≤ 1) : a ≤ b :=
+by simpa only [mul_one] using le_mul_of_le_mul_left h hle
+
+@[to_additive]
+lemma le_mul_of_le_mul_right (h : a ≤ b * c) (hle : b ≤ d) : a ≤ d * c :=
+h.trans (mul_le_mul_right' hle c)
+
+@[to_additive le_of_le_add_right]
+lemma le_of_le_mul_right' (h : a ≤ b * c) (hle : b ≤ 1) : a ≤ c :=
```

Comment:
I had to deviate here from the naming pattern I use since [le_of_le_mul_right](https://leanprover-community.github.io/mathlib_docs/find/le_of_le_mul_right) already exists.